var/spack/repos/builtin/packages/lz4/package.py

```diff
@@ -26,6 +26,9 @@ class Lz4(MakefilePackage):
 
     depends_on('valgrind', type='test')
 
+    variant('shared', default=True, description='Build shared library')
+    variant('static', default=True, description='Build static library')
```

Comment:
Hmm, not convinced of this modeling as `lz4~shared~static` doesn't make much sense. Can we use either a multi-valued variant or just shared as many packages do (and assume `~shared` is equivalent to ask for static)?

Reply:
hm, it maps directly to the autotools flags `--[enable|disable]-static` and `--[enable|disable]-shared]`.

Reply:
actually i think it's better to have 2 flags, so that packages can always to depends_on('lz4 +shared') if they need shared libs, and another can do depends_on('lz4 +static'), and it won't lead to conflicts, whereas depends_on('lz4 ~shared') would be a concretization failure.

Reply:
> and it won't lead to conflicts, whereas depends_on('lz4 ~shared') would be a concretization failure.

Yes, but they would need to conflict with `~shared~static`. I think the best way to model what you say is a multi-valued variant.